geospaas/base_viewer/tests.py

```diff
@@ -0,0 +1,145 @@
+                      ('No datasets are available (or maybe no one is ingested)' in dat) for dat in self.parser.data])
+
+    def test_the_post_verb_of_GUI_without_polygon(self):
+        """shall return the uri of fixtures' datasets in the specified placement of datasets inside the resulted HTML
+        in the case of a POST request without any polygon from user """
+        res3 = self.client.post('/', {
+            'time_coverage_start': timezone.datetime(2000, 12, 29),
+            'time_coverage_end': timezone.datetime(2020, 1, 1),
+            'source': 1})
+        self.assertEqual(res3.status_code, 200)
+        self.parser.feed(str(res3.content))
+        # both datasets must be in the html
+        self.assertIn(True, [('file://localhost/some/test/file1.ext' in dat)
+                             for dat in self.parser.data])
+        self.assertIn(True, [('file://localhost/some/test/file2.ext' in dat)
+                             for dat in self.parser.data])
+
+    def test_the_post_verb_of_GUI_incorrect_dates_without_polygon(self):
+        """shall return 'No datasets are...' in the specified placement of datasets inside the resulted HTML
+        in the case of a POST request with incorrect dates from user and without any polygon from user"""
+        res3 = self.client.post('/', {
+            'time_coverage_start': timezone.datetime(2019, 12, 29),
+            'time_coverage_end': timezone.datetime(2020, 1, 1),
+            'source': 1})
+        self.assertEqual(res3.status_code, 200)
+        self.parser.feed(str(res3.content))
+        self.assertIn(True, [
+                      ('No datasets are available (or maybe no one is ingested)' in dat) for dat in self.parser.data])
+
+    def test_the_get_verb_of_GUI(self):
+        """shall return ALL uri of fixtures' datasets in the specified placement
+        of datasets inside the resulted HTML in the case of a GET request"""
+        res4 = self.client.get('/')
+        self.assertEqual(res4.status_code, 200)
+        self.parser.feed(str(res4.content))
+        # both datasets must be in the html
+        self.assertIn(True, [('file://localhost/some/test/file1.ext' in dat)
+                             for dat in self.parser.data])
+        self.assertIn(True, [('file://localhost/some/test/file2.ext' in dat)
+                             for dat in self.parser.data])
+
+
+class UnitTestingForViews(TestCase):
+    fixtures = ["vocabularies", "catalog"]
+
+    def setUp(self):
+        self.v1 = IndexView()
+        self.ds = self.v1.get_all_datasets()
+        self.forms = self.v1.create_forms(self)
+
+    def test_creating_the_forms(self):
+        '''shall create the specified fields of forms'''
+        self.assertTrue('polygon' in self.forms.__str__())
+        self.assertTrue('time_coverage_start' in self.forms.__str__())
+        self.assertTrue('time_coverage_end' in self.forms.__str__())
+        self.assertTrue('source' in self.forms.__str__())
```

Comment:
Since we switch to Forms the examples above are not valid anymore. An example on how to write a unit test for IndexView. A valid test example is here:
```python
class IndexViewTests(TestCase):
    @patch('geospaas.base_viewer.views.CatalogDataset')
    def test_get_all_datasets(self, mock_dataset):
        """ Shall call CatalogDataset.objects.all() inside get_all_datasets """
        ds = IndexView.get_all_datasets()
        mock_dataset.objects.all.assert_called_once()
```
Please pay attention to the name of the class, name of the method, dockstring. These are quite standard. Don't invent anything else.